spec/integration/active_record_spec.rb

```diff
@@ -97,17 +97,22 @@ def restart_dummy_rails_server
     expect(response.body).to include("writing")
   end
 
-  it "should failover if PG exception is raised before ActionDispatch::DebugExceptions" do
-    flood_get("/trigger-middleware-pg-exception", times: 10) do |response|
-      expect(response.code.to_i).to eq(500)
+  context "when PG exception is raised before ActionDispatch::DebugExceptions" do
+    let(:path) do
+      Pathname.new("#{__dir__}/../support/dummy_app/triggered_from_pg_exception.writing")
     end
 
-    sleep 0.5
-    response = get("/posts")
+    after { FileUtils.rm_f(path) }
```

Comment:
We should try to remove it also `before { }` this spec. Otherwise you could get false-positives when a previous run didn't finish correctly

Reply:
If the documentation is right, then `after` blocks always run no matter what :slightly_smiling_face: 
https://rspec.info/documentation/3.12/rspec-core/RSpec/Core/Hooks.html#after-instance_method